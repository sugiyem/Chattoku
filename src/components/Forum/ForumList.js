import { Alert, SectionList, Text } from "react-native";
import FetchForumData from "./FetchForumData";
import { firebase } from "../../firebase/Config";
import ForumCard from "./ForumCard";
import { useState, useEffect } from "react";
import { Header } from "react-native-elements";

const initialData = [
  {
    data: []
  }
];

const ForumList = () => {
  const [data, setData] = useState(initialData);

  console.log(data);

  useEffect(() => {
    console.log("useEffect triggered");
    return FetchForumData(
      (documents) => setData([{ data: documents }]),
      (e) => console.error(e)
    );
  }, []);

  const renderItem = ({ section, item }) => {
    return (
      <>
        <ForumCard {...item} />
      </>
    );
  };

  const renderHeader = () => <></>;

  const renderFooter = () => <></>;

  return (
    <>
      <Header centerComponent={<Text>List Of Forums</Text>} />
      <SectionList
        removeClippedSubviews={true}
        sections={data}
        renderItem={renderItem}
        renderHeader={renderHeader}
        renderFooter={renderFooter}
      />
    </>
  );
};

export default ForumList;
