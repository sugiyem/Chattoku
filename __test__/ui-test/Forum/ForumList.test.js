import React from "react";
import renderer from "react-test-renderer";
import ForumList from "../../../src/components/Forum/ForumList";

describe("Test Forum List UI", () => {
  it("Renders correctly", () => {
    const tree = renderer.create(<ForumList data={[]} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
