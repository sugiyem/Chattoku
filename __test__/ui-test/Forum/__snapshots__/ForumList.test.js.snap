// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Forum List UI Renders correctly 1`] = `
Array [
  <TextInput
    onChangeText={[Function]}
    placeholder="Search Forum By Name..."
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "whitesmoke",
          "borderColor": "black",
          "borderRadius": 10,
          "borderWidth": 1,
          "color": "black",
          "flexGrow": 0,
          "flexShrink": 0,
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
          "paddingBottom": 5,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 5,
        },
      ]
    }
  />,
  <RCTScrollView
    data={
      Array [
        Object {
          "data": Array [],
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={true}
    renderFooter={[Function]}
    renderHeader={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={
      Array [
        0,
      ]
    }
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      />
      <View
        onLayout={[Function]}
        style={null}
      />
    </View>
  </RCTScrollView>,
]
`;
